CareKit/Connect/OCKContactInfo.h

```diff
@@ -0,0 +1,97 @@
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSInteger, OCKContactInfoType) {
+	OCKContactInfoTypePhone = 0,
+	OCKContactInfoTypeMessage,
+	OCKContactInfoTypeEmail,
+	OCKContactInfoTypeVideo
+};
+
+@interface OCKContactInfo : NSObject <NSSecureCoding, NSCopying>
+
+@property (nonatomic, readonly) OCKContactInfoType type;
+@property (nonatomic, readonly) NSString *displayString;
+@property (nonatomic, readonly) NSURL *actionURL;
+@property (nonatomic, readonly) NSString *label;
+@property (nonatomic, readonly) UIImage *icon;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ Creates a new contact information item. The label and icon will be set to the default for the given contact type.
+ 
+ @param type The type of contact info, which will define the display icon and the default action if no actionURL is specified.
+ @param displayString The string that will be displayed to the user to represent the contact item.
+ @param actionURL The action to take to initiate the connection. For example "sms:1-314-555-1234", "tel:1-314-555-1234", "facetime://user@example.com"
+ 
+ @return An instance of the contact info.
+ */
+- (instancetype)initWithType:(OCKContactInfoType)type displayString:(NSString *)displayString actionURL:(NSURL * _Nullable)actionURL;
+
+/**
+ Creates a new contact information item. The icon will be set to the default for the given contact type.
+ 
+ @param type The type of contact info, which will define the display icon and the default action if no actionURL is specified.
+ @param displayString The string that will be displayed to the user to represent the contact item.
+ @param actionURL The action to take to initiate the connection. For example "sms:1-314-555-1234", "tel:1-314-555-1234", "facetime://user@example.com"
+ @param label The label of the contact info.
+ 
+ @return An instance of the contact info.
+ */
+- (instancetype)initWithType:(OCKContactInfoType)type displayString:(NSString *)displayString actionURL:(NSURL * _Nullable)actionURL label:(NSString *)label;
```

Comment:
Change - (instancetype)initWithType:(OCKContactInfoType)type displayString:(NSString *)displayString actionURL:(NSURL * _Nullable)actionURL to - (instancetype)initWithType:(OCKContactInfoType)type displayString:(NSString *)displayString actionURL:(nullable NSURL *)actionURL.

For consistency.